src/server/index.js

```diff
@@ -205,11 +174,15 @@ class KiteServer extends Emitter {
   }
 
   onConnection(ws) {
-    const proto = dnodeProtocol(this.api)
+    const proto = dnodeProtocol(this.api.methods)
     proto.on('request', this.lazyBound('handleRequest', ws))
 
     const id = ws.getId()
-    ws.on('message', this.lazyBound('handleMessage', proto))
+
+    ws.on('message', message => {
+      handleIncomingMessage.call(this, proto, parse(message))
```

Comment:
and it would be better to parse it on one place to keep `server` and `client` classes as clean as possible.